plugins/Kaleidoscope-Colormap-Overlay/src/kaleidoscope/plugin/Colormap-Overlay.cpp

```diff
@@ -30,14 +30,39 @@
 #include "kaleidoscope/key_defs.h"              // for Key, KEY_FLAGS, Key_NoKey, LockLayer
 #include "kaleidoscope/layers.h"                // for Layer, Layer_
 #include "kaleidoscope/plugin/LEDControl.h"     // for LEDControl
+#include <Kaleidoscope.h>                       // for Kaleidoscope
+#include <Kaleidoscope-FocusSerial.h>           // for Focus
 #include <Kaleidoscope-LED-Palette-Theme.h>     // for LEDPaletteTheme
+#include <Kaleidoscope-EEPROM-Settings.h>       // for EEPROMSettings
 
 namespace kaleidoscope {
 namespace plugin {
 uint16_t ColormapOverlay::map_base_;
+uint16_t ColormapOverlay::overlays_base_;
+uint8_t ColormapOverlay::max_overlays_ = 64;  // TODO(EvyBongers): figure this out. How determine a good maximum?
+Overlay *ColormapOverlay::overlays_;
+uint8_t ColormapOverlay::overlay_count_;
 
 void ColormapOverlay::setup() {
+  // It appears that a call to ::LEDPaletteTheme.reserveThemes() is needed
+  // because it's where palette_base_ gets initialized. Since this plugin (and
+  // possibly others) don't actually use themes, requesting memory for storing
+  // themes doesn't make much sense. Maybe initialisation of palette_base_
+  // could be moved to a setup() method, though maybe the palette should be
+  // split from palette theme altogether?
   map_base_ = ::LEDPaletteTheme.reserveThemes(1);
+
+  overlays_base_ = ::EEPROMSettings.requestSlice(max_overlays_ * sizeof(Overlay));
+
+  ::EEPROMSettings.seal();
+
+  if (!::EEPROMSettings.isValid()) {
+    // TODO(EvyBongers): figure this out. What to do when the settings are out of sync...
+
+    return;
+  }
```

Comment:
I don't really have any idea how to handle this case. I suppose I might change the condition to 'if(isValid()) { /* get overlays from storage */ }', but that doesn't help me with informing the user that there's corruption in EEPROM

Reply:
I think that when eeprom settings isn't valid, it probably makes sense to wipe the config.